### PR TITLE
Update stripe-js for Custom Checkout types

### PR DESCRIPTION
**Pull request by danwang-stripe**
### Summary & motivation
Update version of stripe-js to [v3.0.7](https://github.com/stripe/stripe-js/releases/tag/v3.0.7) to pull in new Custom Checkout types

### API review

Copy [this template] **or** link to an API review issue.

[this template]:
  https://github.com/stripe/react-stripe-js/tree/master/.github/API_REVIEW.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,15 +92,15 @@
     "react-dom": "18.1.0",
     "react-test-renderer": "^18.0.0",
     "rimraf": "^2.6.2",
-    "rollup": "^1.27.0",
+    "rollup": "^4.12.0",
```
Comment (madhav-stripe) [inline package.json:95]: @danwang-stripe I don't think we can upgrade rollup in this PR

**Review by madhav-stripe — CHANGES_REQUESTED** [review]

**Comment by madhav-stripe** [comment]
build output on current master:

```
-rw-r--r--  1 madhav  staff  44871 Mar  5 10:25 react-stripe.d.ts
-rw-r--r--  1 madhav  staff  44871 Mar  5 10:25 react-stripe.esm.d.ts
-rw-r--r--  1 madhav  staff  41327 Mar  5 10:25 react-stripe.esm.js
-rw-r--r--  1 madhav  staff  42399 Mar  5 10:25 react-stripe.js
-rw-r--r--  1 madhav  staff  44871 Mar  5 10:25 react-stripe.umd.d.ts
-rw-r--r--  1 madhav  staff  46976 Mar  5 10:25 react-stripe.umd.js
-rw-r--r--  1 madhav  staff  44871 Mar  5 10:25 react-stripe.umd.min.d.ts
-rw-r--r--  1 madhav  staff  16435 Mar  5 10:25 react-stripe.umd.min.js
```

**Comment by danwang-stripe** [comment]
r? @madhav-stripe 

Merged master in with 776705b085f9daa3c045a0a1da06326266e7d3e7

Thanks for helping with the dependency upgrades!

**Review by madhav-stripe — APPROVED** [review]

**Pull request merged**